libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,308 @@
+-- Output2: front right femur (thigh) servo
+-- Output3: front right tibia (shin) servo
+-- Output5: front left coxa (hip) servo
+-- Output6: front left femur (thigh) servo
+-- Output7: front left tibia (shin) servo
+-- Output8: back right coxa (hip) servo
+-- Output9: back right femur (thigh) servo
+-- Output10: back right tibia (shin) servo
+-- Output11: back left coxa (hip) servo
+-- Output12: back left femur (thigh) servo
+-- Output13: back left tibia (shin) servo
+--
+-- CAUTION: This script should only be used with ArduPilot Rover's firmware
+
+
+local FRAME_LEN = 80    -- frame length in mm
+local FRAME_WIDTH = 150 -- frame width in mm
+
+local COXA_LEN = 30     -- distance (in mm) from coxa (aka hip) servo to femur servo
+local FEMUR_LEN = 85    -- distance (in mm) from femur servo to tibia servo
+local TIBIA_LEN = 125   -- distance (in mm) from tibia servo to foot
+
+--body position and rotation parameters
+local body_rot_max = 10 -- body rotation maximum for any individual axis
+local body_rot_x = 0    -- body rotation about the X axis (i.e. roll rotation)
+local body_rot_y = 0    -- body rotation about the Y axis (i.e. pitch rotation)
+local body_rot_z = 0    -- body rotation about the Z axis (i.e. yaw rotation)
+local body_pos_x = 0    -- body position in the X axis (i.e. forward, back).  should be -40mm to +40mm
+local body_pos_y = 0    -- body position in the Y axis (i.e. right, left).  should be -40mm to +40mm
+local body_pos_z = 0    -- body position in the Z axis (i.e. up, down).  should be -40mm to +40mm
+
+-- starting positions of the legs
+local endpoints1 = {math.cos(45/180*math.pi)*(COXA_LEN + FEMUR_LEN), math.sin(45/180*math.pi)*(COXA_LEN + FEMUR_LEN), TIBIA_LEN }
+local endpoints2 = {math.cos(45/180*math.pi)*(COXA_LEN + FEMUR_LEN), math.sin(-45/180*math.pi)*(COXA_LEN + FEMUR_LEN), TIBIA_LEN }
+local endpoints3 = {-math.cos(45/180*math.pi)*(COXA_LEN + FEMUR_LEN), math.sin(-45/180*math.pi)*(COXA_LEN + FEMUR_LEN), TIBIA_LEN }
+local endpoints4 = {-math.cos(45/180*math.pi)*(COXA_LEN + FEMUR_LEN), math.sin(45/180*math.pi)*(COXA_LEN + FEMUR_LEN), TIBIA_LEN }
+
+-- control input enum
+local control_input_roll = 1
+local control_input_pitch = 2
+local control_input_throttle = 3
+local control_input_yaw = 4
+
+local xy_speed_max = 40     -- x and y axis speed max (used to convert control input) in mm
+local yaw_speed_max = 10    -- yaw speed maximum (used to convert control input)
+local speed_dz = 5          -- speed deadzone.  x, y and yaw speed requests are ignored if their absolute value is less than this number
+local x_speed = 0           -- forward speed target
+local y_speed = 0           -- lateral speed target (right is positive, left is negative)
+local yaw_speed = 0         -- yaw rotation speed target
+
+local leg_lift_height = 50  -- leg lift height (in mm) while walking
+
+-- gait definition parameters
+local gait_type = 0         -- gait pattern.  0 = alternating gait, 1 = wave gait.
```

Comment:
So from a style point of view you'd normally end up starting to lump these into a table with keys of `step`, `step_total` etc, rather then have a local for each. Not a huge deal, and theirs some arguments either way, but that can sometimes help clean up and encapsulate the code better.

Reply:
yes, I wanted to put these into a structure but didn't know how to so I just made all the variables start with the "gait_" prefix.  I've read up a bit on tables and I think we can move them into one although maybe we'd best do that as part of a follow-up PR.

Reply:
I've added this to the to-do list as well. https://github.com/ArduPilot/ardupilot/issues/15112